ui/app/utils/json-with-default.js

```diff
@@ -0,0 +1,10 @@
+import { copy } from '@ember/object/internals';
+
+// Used with fetch.
+// Fetch only goes into the promise catch if there is a network error.
+// This means that handling a 4xx or 5xx error is the responsibility
+// of the developer.
+const jsonWithDefault = defaultResponse => res =>
+  res.ok ? res.json() : copy(defaultResponse, true);
+
```

Comment:
This is nice, I just realized it depends on the `res` having an `ok` property. So, tiny super nit here, but would naming it differently or putting it in a subfolder to make it clearer that the `res` needs an `ok` property and `json` method.

Isn't that a `Response`?

https://developer.mozilla.org/en-US/docs/Web/API/Response/ok

So maybe `responseWithDefault` or maybe put it in `utils/http/response/...` or similar?

The commenting here helps, but if I'm in a file that uses this I might think that this just requires a json blob, whereas it needs a `Response`

Reply:
Yeah, I considered this interface gotcha, but decided ultimately that I was probably overthinking it, and it isn't as big of a footgun as it seems under a microscope. 

If it comes up again, I'll considering moving it.